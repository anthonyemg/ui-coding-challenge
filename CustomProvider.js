import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from './src/public/reducers';

const store = createStore(reducer);

const CustomProvider = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};

export default CustomProvider;
