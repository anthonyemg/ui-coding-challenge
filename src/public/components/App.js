import React from 'react';
import { connect } from 'react-redux';
import { updateName, updateAddress, updateFavorites } from '../actions';
import { HashRouter, Route } from 'react-router-dom';
import Landing from './Landing';

class App extends React.Component {
  constructor(props) {
    super(props);
  }
  componentWillMount() {
    fetch('/database', { method: 'GET' })
      .then(res => res.json())
      .then(res => {
        this.props.updateName(res.name);
        this.props.updateAddress(res.address);
        this.props.updateFavorites(res.favorites);
      })
      .catch(err => console.log(err));
  }
  render() {
    return (
      <HashRouter>
        <div className="app">
          <Route exact path="/" component={Landing} />
        </div>
      </HashRouter>
    );
  }
}
const mapStateToProps = state => ({});
const mapDispatchToProps = dispatch => ({
  updateName: name => dispatch(updateName(name)),
  updateAddress: address => dispatch(updateAddress(address)),
  updateFavorites: favorites => dispatch(updateFavorites(favorites))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
