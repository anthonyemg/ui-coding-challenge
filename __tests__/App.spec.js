import React from 'react';
import renderer from 'react-test-renderer';
import CustomProvider from '../CustomProvider';
import App from '../src/public/components/App';

test('the App component renders correctly', () => {
  const component = renderer.create(
    <CustomProvider>
      <App />
    </CustomProvider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
