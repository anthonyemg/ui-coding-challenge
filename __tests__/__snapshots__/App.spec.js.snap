// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the App component renders correctly 1`] = `
<div
  className="app"
>
  <div
    className="landing-container"
  >
    <div
      className="landing-header"
    >
      Motorcycle Magazine
    </div>
    <span
      className="landing-settingsTitle"
    >
      Motorcycle Magazine Settings
    </span>
    <div
      className="component-container"
    >
      <div
        className="component-wrapper"
      >
        <div
          className="component-details"
        >
          <span
            className="component-title"
          >
            Name
          </span>
          <span
            className="component-text"
          >
            Anthony G
          </span>
        </div>
        <div>
          <button
            className="component-editButton"
            onClick={[Function]}
          >
            Edit Name
          </button>
        </div>
      </div>
    </div>
    <div
      className="component-container"
    >
      <div
        className="component-wrapper"
      >
        <div
          className="component-details"
        >
          <span
            className="component-title"
          >
            Address
          </span>
          <span
            className="component-text"
          >
            123 W Ave
          </span>
          <span
            className="component-text"
          >
            NY, NY 10000
          </span>
        </div>
        <div>
          <button
            className="component-editButton"
            onClick={[Function]}
          >
            Edit Address
          </button>
        </div>
      </div>
    </div>
    <div
      className="component-container"
    >
      <div
        className="component-wrapper"
      >
        <div
          className="component-details"
        >
          <span
            className="component-title"
          >
            Favorite Motorcycles
          </span>
          <span
            className="component-noneAdded"
          >
            None Added
          </span>
        </div>
        <div>
          <button
            className="component-editButton"
            onClick={[Function]}
          >
            Edit Motorcyles
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
